### PR TITLE
Enhancement - Add href support to FileDetails download button

### PR DESCRIPTION
**Pull request by lukemojo**
Add href and target props to FileDetails
Add story
Add test
Update types

Closes [VE-91](https://sce.myjetbrains.com/youtrack/issue/VE-91/Add-href-support-to-the-FileDetail-component)
Contributes to [TD-2863](https://sce.myjetbrains.com/youtrack/issue/TD-2863/Handle-api-errors-when-downloading-roads)

## Changes

Adds href support to the FileDetails component.

This introduces
- a downloadLinkHref prop
- a downloadLinkTarget prop
- Replaces the button with an anchor when the href is provided

## Testing notes

Check the file detail story that the button renders as an anchor and passes through the href. Ensure the other stories function with no change.

## Author checklist before assigning a reviewer

- [x] Reviewed my own code-diff.
- [x] PR assigned to me or an appropriate delegate.
- [x] Relevant labels added to the PR.
- [x] Appropriate tests have been added.
- [x] Lint and test workflows pass.

### CODE REVIEW

File: src/CardContent/FileDetails/FileDetails.types.ts [hunk 1]
```diff
@@ -9,6 +9,14 @@ export type FileDetailsProps = {
...
    */
   downloadButtonTextOnSearch?: string;
+  /**
+   * The download href link.
+   */
+  downloadLinkHref?: string;
+  /**
+   * The download href link target attribute.
+   */
+  downloadLinkTarget?: string;
```
Comment (mattcorner) [inline src/CardContent/FileDetails/FileDetails.types.ts:19]: Is there a use case for target? I understood that SCENE would open the download url and then redirect you back to the current page. When would we want that to be in a new tab?

Just checking we're not adding unnecessary support here.

File: src/CardContent/FileDetails/FileDetails.types.ts [hunk 2]
```diff
@@ -9,6 +9,14 @@ export type FileDetailsProps = {
    * The download button text on search.
    */
   downloadButtonTextOnSearch?: string;
+  /**
+   * The download href link.
+   */
+  downloadLinkHref?: string;
```
Comment (mattcorner) [inline src/CardContent/FileDetails/FileDetails.types.ts:15]: I think a lesson for next time we write a component like this is that nested props are a good idea. e.g. downloadLinkProps: { text, href, target, onClick} etc.

Too late to change now.

File: src/CardContent/FileDetails/FileDetails.types.ts (hunk 1 above)
Comment (mattcorner) [inline src/CardContent/FileDetails/FileDetails.types.ts:19]: I don't think string is the correct type here. It's actually a string union. Seems like react has a `HTMLAttributeAnchorTarget` type somewhere we should presumably be able to access.

image

**Review by mattcorner — CHANGES_REQUESTED** [review]

File: src/CardContent/FileDetails/FileDetails.types.ts (hunk 1 above)
Comment (lukemojo) [reply src/CardContent/FileDetails/FileDetails.types.ts]: Perhaps. I figured if we're giving href support and using an anchor then we should give basic attribute support. I don't think they will be using it for the current story.

**Review by mattcorner — APPROVED** [review]

**Pull request merged**